### DISCUSSION TITLE
Shadow External Command

### DISCUSSION THREAD

**Discussion by theAkito** [nushell/nushell#6998] [Q&A] [answered]
When trying to shadow an external command, the shell still uses the external command.

### Example

```nu
export alias hostname = (sys | get host.hostname)
```

When this is at the bottom of my `config.nu` and the file is sourced, then `hostname` still uses the external `/bin/hostname`.

How do I make this work?

**Comment by theAkito** [comment]
I found out, that this is actually related to `source` not working anymore. I used it tons of times to source `env.nu` and `config.nu` manually. Now, it stopped working. The files are only sourced on `nu` launch, but `source` manually does not work anymore.

**Comment by theAkito** [comment, marked as answer]
I see the problem now.

This works:
```nu
export alias sauce = source /home/user/.config/nushell/config.nu
```

But this does not work:
```nu
export alias sauce = (source /home/user/.config/nushell/env.nu; source /home/user/.config/nushell/config.nu)
```

Leaving out the paranthesis does not even make the alias work, at all.

**Comment by kubouch** [comment]
```
export alias sauce = source /home/user/.config/nushell/config.nu
```
should work the same way
```
alias sauce = source /home/user/.config/nushell/config.nu
```

Can you file a bug report?

The reason why the `()` does not seem to work is because the sourced content is scoped within the subexpression. Whatever you set in the config will be scoped inside the `()`.

**Reply by theAkito** [reply]
> The reason why the `()` does not seem to work is because the sourced content is scoped within the subexpression. Whatever you set in the config will be scoped inside the `()`.

Ah, yes, that makes sense!

> should work the same way

Not sure, what exactly you are referring to.

> Can you file a bug report?

Yes, though what would be the expected behaviour now? Since paranthesis are supposed to not work in this example, if I understood correctly.

**Reply by kubouch** [reply]
`export` alias should work identically to `alias` when sourced. The `export` keyword makes sense only in the context of modules.

**Reply by theAkito** [reply]
It isn't sourced though. The alias is in a module, which is `use`d by `config.nu`. Then, `config.nu` is sourced and I get the `alias` from the module, which `config.nu` is `use`ing.
